packages/top-nav/stories/top-nav.stories.ts

```diff
@@ -106,3 +107,44 @@ export const Selected = (): TemplateResult => {
         </sp-top-nav>
     `;
 };
+
+// https://spectrum.adobe.com/page/application-frame/#Application-mode
+// https://spectrum.adobe.com/page/headers/#Anatomy
+
+export const Modes = (): TemplateResult => {
+    return html`
+        <sp-top-nav>
+            <sp-action-button quiet href="/marketing-page" label="App">
+                <sp-icon slot="icon" src=${logo}></sp-icon>
+            </sp-action-button>
+            <sp-top-nav-item href="#">Design</sp-top-nav-item>
+            <sp-top-nav-item href='#prototype'>Prototype</sp-top-nav-item>
+            <span style="margin-inline-start: auto; font-size: 1.1em;">document_name.pdf</span>
+            <sp-button-group style="margin-inline-start: auto;">
+                <sp-action-button quiet label="Search">
+                    <sp-icon size="s" slot="icon">${SearchIcon()}</sp-icon>
+                </sp-action-button>
+                <sp-action-button quiet label="Download">
+                    <sp-icon size="s" slot="icon">${DownloadIcon()}</sp-icon>
+                </sp-action-button>
+                <sp-action-button quiet label="Link">
+                    <sp-icon size="s" slot="icon">${LinkIcon()}</sp-icon>
+                </sp-action-button>
+            </sp-button-group>
+            <sp-action-menu label="User" style="margin-inline-start: auto;">
```

Comment:
This is something we'd want to review deeper with the Spectrum team before codifying as practice, but I think the margin here is not expected. With the move to `sp-action-group` above, you can include the `sp-action-menu` in that parent for the spacing that seems to be outlined in the source spec.

Reply:
Aha! Much better, thanks. I tried that with `sp-button-group` and got inconsistent margins, but `sp-action-group` delivers the layout I expected.